### PR TITLE
Make Xunit detect tests from Xml custom test fx (no more outerloop tests in S.P.Xml)

### PR DESCRIPTION
**Pull request by krwq**
This improves debuggability of XML tests. Before this PR when one of the test cases failed you couldn't tell which one (one test case was a batch of hundreds to thousands tests). This PR makes it so that XML custom test framework is feeding Xunit's theory with inline data which is enough to execute test and show description of the test case - this way we can nicely see a failure
Fixes https://github.com/dotnet/corefx/issues/14855
Contributes to https://github.com/dotnet/corefx/issues/3133

### CODE REVIEW

**Comment by danmosemsft** [comment]
Good to see tests moving to inner loop, thanks!

**Comment by krwq** [comment]
@mellinoe could you take a look at https://mc.dot.net/#/user/krwq/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/1efef1283c0c3338b2adcbb70a54780b11b9340e ? This doesn't seem to be related to this PR

**Comment by krwq** [comment]
test Linux x64 Release Build

**Comment by krwq** [comment]
test OSX x64 Debug Build

File: src/Common/tests/System/Xml/ModuleCore/XunitRunner.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+﻿using System;
```
Comment (pjanotti) [inline src/Common/tests/System/Xml/ModuleCore/XunitRunner.cs]: q: don't we add the boiler plate header to test files?

File: src/Common/tests/System/Xml/ModuleCore/XunitRunner.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        }
+
+        public virtual IEnumerable<object[]> GetData(IAttributeInfo dataAttribute, IMethodInfo testMethod)
+        {
+            string methodName = (string)dataAttribute.GetConstructorArguments().Single();
+            Func<CTestModule> moduleGenerator = XmlTestsAttribute.GetGenerator(GetDeclaringType(testMethod), methodName);
+            return GenerateTestCases(moduleGenerator);
+        }
+
+        public virtual bool SupportsDiscoveryEnumeration(IAttributeInfo dataAttribute, IMethodInfo testMethod)
```
Comment (pjanotti) [inline src/Common/tests/System/Xml/ModuleCore/XunitRunner.cs]: nit: I like expression-body method for this case

**Comment by krwq** [comment]
test Linux x64 Release Build

**Comment by krwq** [comment]
test Windows x86 Release Build

**Comment by krwq** [comment]
Merging since Linux failure is unrelated (System.Drawing failure)

**Pull request merged**